sys/fs/cd9660/cd9660_vnops.c

```diff
@@ -140,8 +144,27 @@ cd9660_access(struct vop_access_args *ap)
 		}
 	}
 
-	return (vaccess(vp->v_type, ip->inode.iso_mode, ip->inode.iso_uid,
-	    ip->inode.iso_gid, ap->a_accmode, ap->a_cred));
+	file_mode = ip->inode.iso_mode;
+	switch (vp->v_type) {
+	case VDIR:
+		if (ip->i_mnt->im_flags & ISOFSMNT_DMASK)
+			file_mode = ip->i_mnt->im_dmask;
```

Comment:
Should this be `file_mode &= ip->i_mnt->im_dmask` instead of an assignment?  msdosfs uses the &

Reply:
This part is enforcing the dmask passed as option.

Reply:
Fixed.